apps/store/src/features/carDealership/ActionButtonsCar.tsx

```diff
@@ -2,13 +2,18 @@ import { datadogRum } from '@datadog/browser-rum'
 import styled from '@emotion/styled'
 import { useState } from 'react'
 import { theme } from 'ui'
+import { ActionButton } from '@/components/ProductItem/ProductItem'
 import { type ProductOfferFragment } from '@/services/apollo/generated'
 import { PriceIntent } from '@/services/priceIntent/priceIntent.types'
-import { ActionButton } from '../../components/ProductItem/ProductItem'
 import { ActionStateEdit } from './ActionStateEdit'
 import { useEditAndConfirm } from './useEditAndConfirm'
 
-type State = 'IDLE' | 'EDITING'
+type State = { type: 'IDLE' } | { type: 'EDITING' } | { type: 'SUBMITTING'; tierLevel?: string }
+
+const STATE: Record<string, State> = {
+  IDLE: { type: 'IDLE' },
+  EDITING: { type: 'EDITING' },
+} as const
```

Comment:
nit
```suggestion
const STATE: Record<'IDLE' | 'EDITING', State> = {
  IDLE: { type: 'IDLE' },
  EDITING: { type: 'EDITING' },
}
```
Just a suggestion. I find more explicit and easy to follow. It's not flexible for changes tho.

Reply:
hmm yeah I understand, perhaps I can derive the type and use as key

although the keys don't really have anything to do with the state type, right?

In the sense I can setup multiple keys pointing to the same state type for example